src/green_recognition/src/CircleDetection.cpp

```diff
@@ -0,0 +1,133 @@
+/*
+ * Created By: Robyn Castro
+ * Created On: June 17, 2017
+ * Description: Determines whether or not green is seen on
+ *              the screen then publishes a recommended twist
+ *              message.
+ */
+
+#include <CircleDetection.h>
+#include <image_transport/image_transport.h>
+#include <std_msgs/Bool.h>
+
+using namespace cv;
+using namespace std;
+using namespace cv_bridge;
+
+GreenRecognition::GreenRecognition(std::string &image_path) {
+
+    cv::Mat bgr_image = imread(image_path);
+
+    // Check if the image can be loaded
+    check_if_image_exist(bgr_image, image_path);
+
+    minTargetRadius = 20;
+    countCircles(bgr_image);
+
+    waitKey(0);
+
+}
+
+GreenRecognition::GreenRecognition() {
+    minTargetRadius = 20;
+}
+
+GreenRecognition::GreenRecognition(int argc, char **argv, std::string node_name) {
+
+    // Setup handles
+    ros::init(argc, argv, node_name);
+    ros::NodeHandle nh;
+    ros::NodeHandle private_nh("~");
+
+    // Setup image transport
+    image_transport::ImageTransport it(nh);
+
+    // Setup subscriber
+    std::string image_topic = "/robot/vision/filtered_image";
+    int refresh_rate = 10;
+    image_sub = it.subscribe<GreenRecognition>(image_topic, refresh_rate,
+                                               &GreenRecognition::subscriberCallBack, this);
+
+    // Setup publishers
+    std::string output_topic = "/robot/vision/green_detected";
+    uint32_t queue_size = 1;
+    boolean_pub = private_nh.advertise<std_msgs::Bool>(output_topic, queue_size);
+
+    // Get some params
+    SB_getParam(private_nh, "minimum_target_radius", minTargetRadius, 50);
+    SB_getParam(private_nh, "show_image_window", showWindow, true);
+
+}
+
+void GreenRecognition::subscriberCallBack(const sensor_msgs::Image::ConstPtr &image) {
+
+    // If something is seen tell the robot to move
+    int numCircles = countCircles(rosToMat(image));
+    std_msgs::Bool circle_detected;
+    circle_detected.data = numCircles > 0;
+    boolean_pub.publish(circle_detected);
+}
+
+Mat GreenRecognition::rosToMat(const sensor_msgs::Image::ConstPtr &image) {
+    CvImagePtr imagePtr;
+    imagePtr = toCvCopy(image, image->encoding);
+    return imagePtr->image;
+}
+
+int GreenRecognition::countCircles(const Mat &filtered_image) {
+
+    vector<vector<Point>> contours;
+    vector<Vec4i> hierarchy;
+    vector<cv::Point2i> center;
+    vector<int> radii;
+
+    // Convert grayscale image to black and white image
+    cv::Mat bwImage;
+    try {
+        cv::cvtColor(filtered_image, bwImage, CV_RGB2GRAY);
+    } catch(cv::Exception) {
+        bwImage = filtered_image;
+    }
+    cv::findContours(bwImage.clone(), contours, hierarchy, CV_RETR_TREE, CV_CHAIN_APPROX_NONE);
```

Comment:
A quick comment on what we're actually searching for in the image here would be helpful